### ISSUE TITLE
Error when current directory name ends with .go

### ISSUE THREAD

**Issue by scottjbarr** [cweill/gotests#46]
If the name of the directory ends in `.go`, tests cannot be generated.

```
$ pwd
$GOPATH/src/github.com/scottjbarr/foo.go

$ gotests -all .
Parser.Parse source file: ioutil.ReadFile: read $GOPATH/src/github.com/scottjbarr/foo.go: is a directory
```

*I've replaced the actual $GOPATH with `$GOPATH`*

If the name of the directory is changed from "foo.go" to "foo", then test generation works as expected.

**Comment by cweill** [comment]
This seems like an unusual edge case, as I believe directories don't usually end in ".go". Feel free to make a pull request to address this issue.

**Comment by scottjbarr** [comment]
Yes it is odd. Unfortunately I have seen it out in the world at two separate companies now.
